src/internal/components/QrCode/QrCodeSvg.tsx

```diff
@@ -0,0 +1,183 @@
+import { useMemo } from 'react';
+import { useOnchainKit } from '../../../core-react/useOnchainKit';
+import {
+  GRADIENT_END_COORDINATES,
+  GRADIENT_START_COORDINATES,
+  QR_CODE_SIZE,
+  QR_LOGO_BACKGROUND_COLOR,
+  QR_LOGO_RADIUS,
+  QR_LOGO_SIZE,
+  linearGradientStops,
+  ockThemeToLinearGradientColorMap,
+  ockThemeToRadiamGradientColorMap,
+  presetGradients,
+} from './gradientConstants';
+import { useCorners } from './useCorners';
+import { useDotsPath } from './useDotsPath';
+import { useLogo } from './useLogo';
+import { useMatrix } from './useMatrix';
+
+function coordinateAsPercentage(coordinate: number) {
+  return `${coordinate * 100}%`;
+}
+
+export type QRCodeSVGProps = {
+  value: string;
+  size?: number;
+  backgroundColor?: string;
+  logo?: React.ReactNode;
+  logoSize?: number;
+  logoBackgroundColor?: string;
+  logoMargin?: number;
+  logoBorderRadius?: number;
+  quietZone?: number;
+  quietZoneBorderRadius?: number;
+  ecl?: 'L' | 'M' | 'Q' | 'H';
+  gradientType?: 'radial' | 'linear';
+};
+
+export function QrCodeSvg({
+  value,
+  size = QR_CODE_SIZE,
+  backgroundColor = '#ffffff',
```

Comment:
should these default to theme colors to support dark mode?

Reply:
eventually yes. unfortunately, enough of the code here assumes lightmode that getting to dark mode compatible was out of scope. this is what it currently looks like if we try to use the darkmode backgrounds:

<img width="275" alt="image" src="https://github.com/user-attachments/assets/ad77a7c5-afa2-4b32-bc94-51edf4f00ccf" />

dark-mode compatibility is going to be a fast-follow